### PR TITLE
feat(23.10): add grep slice

### PR DESCRIPTION
**Pull request by lwnexgen**
Add `grep` slice for Ubuntu 23.10.

Supports running `grep`, `egrep`, etc.

### CODE REVIEW

File: slices/grep.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package: grep
+
+slices:
+  bins:
+    essential:
+      - libc6_libs
+      - libpcre3_libs
+    contents:
+      /bin/grep:
+  deprecated:
```
Comment (cjdcordeiro) [inline slices/grep.yaml:10]:
```suggestion

  deprecated:
```

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
thanks! also missing `libpcre3` here

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
Sorry, I didn't mean `libpcre3` was missing from the `deprecated` slice, but rather it is missing from the `ubuntu-23.10` release. So if you try to install your slices, it won't work.

But as @rebornplusplus mentioned in https://github.com/canonical/chisel-releases/pull/118#discussion_r1480906803, in 23.10, `grep` depends on libpcre2-8-0 and not libpcre3 (and the former already exists in this chisel ubuntu release)

**Comment by lwnexgen** [comment]
> But as @rebornplusplus mentioned in https://github.com/canonical/chisel-releases/pull/118#discussion_r1480906803, in 23.10, grep depends on libpcre2-8-0 and not libpcre3 (and the former already exists in this chisel ubuntu release)

Whoops, good catch - will resolve and actually test this time.

**Review by cjdcordeiro — APPROVED** [review]
thanks! lgtm

**Review by rebornplusplus — APPROVED** [review]
Looks good to me.

**Pull request merged**